src/org/labkey/test/tests/list/ListDateAndTimeTest.java

```diff
@@ -0,0 +1,1434 @@
+    {
+        ListDateAndTimeTest init = (ListDateAndTimeTest)getCurrentTest();
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+        log("Setup project and list module");
+        _containerHelper.createProject(PROJECT_NAME, null);
+        goToProjectHome();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+    }
+
+    @AfterClass
+    public static void resetAfterClass()
+    {
+        ListDateAndTimeTest init = (ListDateAndTimeTest) getCurrentTest();
+        init.resetSiteSettings();
+    }
+
+    public void resetSiteSettings()
+    {
+        log("Reset site settings.");
+        LookAndFeelSettingsPage.beginAt(this).reset();
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+        if (isElementPresent(PortalHelper.Locators.webPartTitle("Search")))
+            new PortalHelper(this).removeWebPart("Search");
+    }
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    private SimpleDateFormat _defaultDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    private SimpleDateFormat _defaultTimeFormat = new SimpleDateFormat("HH:mm:ss");
+    private SimpleDateFormat _defaultDateTimeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+
+    private void prepForDateAndTimeOnlyTest(String listName) throws IOException, CommandException
+    {
+        log("Reset site setting to have default formats for date and time values.");
+        LookAndFeelSettingsPage settingsPage = LookAndFeelSettingsPage.beginAt(this);
+        settingsPage.reset();
+
+        // Reset doesn't wait for page (page may or may not refresh).
+        sleep(500);
+
+        // Use java's Date object and SimpleDateFormatter (not strings) to enter the data.
+        _defaultDateFormat = new SimpleDateFormat(settingsPage.getDefaultDateDisplay());
+        _defaultTimeFormat = new SimpleDateFormat(settingsPage.getDefaultTimeDisplay());
+        _defaultDateTimeFormat = new SimpleDateFormat(settingsPage.getDefaultDateTimeDisplay());
+
+        deleteListIfPresent(getProjectName(), listName);
+
+    }
+
+    private void deleteListIfPresent(String projectName, String listName) throws IOException, CommandException
+    {
+
+        Connection cn = createDefaultConnection();
+        SelectRowsCommand cmd = new SelectRowsCommand("ListManager", "ListManager");
+        cmd.setColumns(Arrays.asList("Name", "Container"));
+        cmd.addFilter("Container/DisplayName", projectName, Filter.Operator.getOperator("EQUAL"));
+        cmd.addFilter("Name", listName, Filter.Operator.getOperator("EQUAL"));
+        SelectRowsResponse response = cmd.execute(cn, projectName);
+
+        if(response.getRowCount().intValue() != 0)
+        {
+            _listHelper.beginAtList(projectName, listName);
+            _listHelper.deleteList();
+        }
```

Comment:
There's a helper for this.
```suggestion
        DomainUtils.ensureDeleted(projectName, "lists", listName);
```

Reply:
Changed.